### PR TITLE
PHOENIX-6431 Add support for auto assigning pmfs

### PR DESCRIPTION
**Pull request by jpisaac**

### CODE REVIEW

**Comment by jpisaac** [comment]
@yanxinyi @ChinmaySKulkarni

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   7m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 5 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 54s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 37s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  4.x passed  |
| +0 :ok: |  spotbugs  |   1m  0s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 37s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 14s |  phoenix-pherf: The patch generated 27 new + 184 unchanged - 9 fixed = 211 total (was 193)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  5s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  10m 38s |  phoenix-pherf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  45m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1190 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 15dee7c35912 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 5c57004 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/1/testReport/ |
| Max. process+thread count | 2110 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 5 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 48s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 38s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  4.x passed  |
| +0 :ok: |  spotbugs  |   1m  0s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 38s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 14s |  phoenix-pherf: The patch generated 27 new + 184 unchanged - 9 fixed = 211 total (was 193)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  5s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 28s |  phoenix-pherf in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  39m  6s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.pherf.workload.mt.tenantoperation.TenantViewOperationWorkloadIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1190 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 12cdd94fa17a 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 5c57004 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/2/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/2/artifact/yetus-general-check/output/patch-unit-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/2/testReport/ |
| Max. process+thread count | 2149 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-pherf/src/main/java/org/apache/phoenix/pherf/configuration/LoadProfile.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 package org.apache.phoenix.pherf.configuration;
 
+import org.apache.phoenix.pherf.workload.mt.Operation;
```
Comment (yanxinyi) [inline phoenix-pherf/src/main/java/org/apache/phoenix/pherf/configuration/LoadProfile.java:21]: unused import?

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  1s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 5 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 23s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 37s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  4.x passed  |
| +0 :ok: |  spotbugs  |   1m  3s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 38s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 14s |  phoenix-pherf: The patch generated 26 new + 184 unchanged - 9 fixed = 210 total (was 193)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  5s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 56s |  phoenix-pherf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |  38m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1190 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 433544a9b687 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / d962363 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/3/testReport/ |
| Max. process+thread count | 2100 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 5 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m  9s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 37s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  4.x passed  |
| +0 :ok: |  spotbugs  |   0m 59s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 37s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 13s |  phoenix-pherf: The patch generated 26 new + 128 unchanged - 9 fixed = 154 total (was 137)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  6s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 24s |  phoenix-pherf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  38m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1190 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 8a53fc36e85c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / d962363 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/4/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/4/testReport/ |
| Max. process+thread count | 2159 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1190/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**